File: cn/java-spring-demo/src/main/java/com/itangcent/springboot/demo/common/model/TestJsonFieldBean.java
package com.itangcent.springboot.demo.common.model;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * JSON字段名称映射测试Bean
 * 用于演示Jackson的@JsonProperty注解功能
 * 实现Java字段名和JSON字段名的自定义映射
 */
public class TestJsonFieldBean {
    /**
     * 用户名
     * JSON序列化时映射为"user-name"
     */
    @JsonProperty("user-name")
    private String userName;

    /**
     * 用户年龄
     * JSON序列化时映射为"user-age"
     */
    @JsonProperty("user-age")
    private Integer userAge;

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public Integer getUserAge() {
        return userAge;
    }

    public void setUserAge(Integer userAge) {
        this.userAge = userAge;
    }
} 

File: en/kotlin-spring-demo/src/main/kotlin/com/itangcent/common/model/TestJsonFieldBean.kt
package com.itangcent.common.model

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * JSON Field Name Mapping Test Bean
 * Used to demonstrate Jackson's @JsonProperty annotation functionality
 * Implements custom mapping between Java field names and JSON field names
 */
data class TestJsonFieldBean(
    /**
     * Username
     * Mapped to "user-name" in JSON serialization
     */
    @JsonProperty("user-name")
    var userName: String? = null,

    /**
     * User Age
     * Mapped to "user-age" in JSON serialization
     */
    @JsonProperty("user-age")
    var userAge: Int? = null
) 